packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -621,7 +622,47 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         containerScope: FluidObject = context.scope,
         existing?: boolean,
     ): Promise<ContainerRuntime> {
-        // If taggedLogger exists, use it. Otherwise, wrap the vanilla logger:
+        let existingFlag = true;
+        if (!existing) {
+            existingFlag = false;
+        }
+        return this.loadRuntime({
+            context,
+            registryEntries,
+            existing: existingFlag,
+            requestHandler,
+            runtimeOptions,
+            containerScope,
+        });
+    }
+
+    /**
+     * Load the stores from a snapshot and returns the runtime.
+     * @param context - Context of the container.
+     * @param registryEntries - Mapping to the stores.
+     * @param existing - When loading from an existing snapshot
+     * @param requestHandler - Request handlers for the container runtime
+     * @param runtimeOptions - Additional options to be passed to the runtime
+     */
+     public static async loadRuntime(
+        params: {
+            context: IContainerContext;
+            registryEntries: NamedFluidDataStoreRegistryEntries;
+            existing: boolean;
+            requestHandler?: (request: IRequest, runtime: IContainerRuntime) => Promise<IResponse>;
+            runtimeOptions?: IContainerRuntimeOptions;
+            containerScope?: FluidObject;
+        },
+    ): Promise<ContainerRuntime> {
+        const { context, registryEntries, existing, requestHandler } = params;
+        let { runtimeOptions, containerScope } = params;
```

Comment:
Nit: I like to avoid `let` where possible to prevent code later on from changing the values again.

So I would do `const runtimeOptions = params.runtimeOptions ?? {}` and likewise for `containerScope`.

Reply:
Looks like you can do this:

```ts
        const { context, registryEntries, existing, requestHandler, runtimeOptions = {}, containerScope = {} } = params;
```

See [Playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBANiA5jAvDUkRwKYDoGIDcAUMQGYCuYwUAluDGSCABQAOAhgE4cC2EALhgBvGByHQutMERgAjAPxCwFXnOxcYAXwCUI4jEPpw0EWIA081DACMATjvbrnHvxJH4SFh0tydJLVImVmFxGAAiDggAEzJw7R0gqhp6MEZmACYWUR8rNAdtIRyJKCkZQnklGBU1DQT9DwJvX39iQPJM7LDImLiE4iA).  You can even put that object spreading with default values in the function signature itself instead of ever declaring `params`.  But IMO that's pretty verbose for the function signature, unless you name the type so you can refer to it by name there.  Up to you what you do, just sharing options :)